File: JavaGuiHttpReq/LekerApi/src/Kolcsonzok.java
import java.util.ArrayList;

public class Kolcsonzok {
    Integer kolcsonzo_id;
    String nev;
    String szuletesi_datum;
    ArrayList<Kolcsonzesek> kolcsonzesek;

    public Kolcsonzok(
        Integer kolcsonzo_id,
        String nev,
        String szuletesi_datum,
        ArrayList<Kolcsonzesek> kolcsonzesek
    ) {
        this.kolcsonzo_id = kolcsonzo_id;
        this.nev = nev;
        this.szuletesi_datum = szuletesi_datum;
        this.kolcsonzesek = kolcsonzesek;
    }
}

File: api/konyvtar_api/app/Models/Kolcsonzok.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Kolcsonzok extends Model
{
    use HasFactory;

    protected $table = "kolcsonzok";
    protected $primaryKey = "kolcsonzo_id";

    public $timestamps = false;

    protected $fillable =[
        "nev",
        "szuletesi_datum"
    ];

    
    public function kolcsonzesek(){                                 //ebben a táblában lévö elsödleges kulcs primaray key             
        return $this->hasMany(Kolcsonzesek::class, "kolcsonzo_id", "kolcsonzo_id");
    }                                       //kivülre hivatkozo kulcs
}
